_data/data.js

```diff
@@ -213,9 +213,11 @@ const downloadImage = async function (url, file) {
   }
 
   const filename = `${file}${extension}`;
+  let hashCode = '';
   const path = Path.join(IMAGE_DIRECTORY, filename);
   const virtualPath = Path.join(IMAGE_PATH, filename);
 
+  // No longer hits any file since we use hash as file name
```

Comment:
Good catch! Please move the cache miss checker to after the `hashCode` is generated. If a file with the same `hashCode` doesn't exist before, then mark `cacheMissed = true`.